GitUI/CommandsDialogs/ToolStripPushButton.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Windows.Forms;
+using GitCommands;
+using GitUI.Properties;
+using GitUIPluginInterfaces;
+using ResourceManager;
+
+namespace GitUI.CommandsDialogs
+{
+    public class ToolStripPushButton : ToolStripButton
+    {
+        private readonly TranslationString _push = new TranslationString("Push");
+
+        private readonly TranslationString _aheadCommitsToPush =
+            new TranslationString("{0} new commit(s) will be pushed");
+
+        private readonly TranslationString _behindCommitsTointegrateOrForcePush =
+            new TranslationString("{0} commit(s) should be integrated (or will be lost if force pushed)");
+
+        private AheadBehindHelper _aheadBehindHelper;
+
+        public void Initialize(IGitModule gitModule)
+        {
+            _aheadBehindHelper = new AheadBehindHelper(gitModule);
+        }
+
+        public void DisplayAheadBehindInformation()
+        {
+            if (!GitVersion.Current.SupportAheadBehindData)
```

Comment:
This makes the class untestable, let's inject the flag via `Initialize` method like so:
```
        public void Initialize(IGitModule gitModule, bool supportsAheadBehindData)
        {
            ...
        }
```